pkg/controller/controller_instance.go

```diff
@@ -206,202 +206,16 @@ func (c *controller) reconcileServiceInstanceDelete(instance *v1alpha1.ServiceIn
 	// If there is no op in progress, and the instance was never provisioned,
 	// we can just delete. this can happen if the service class name
 	// referenced never existed.
-	//
-	// TODO: the above logic changes slightly once we handle orphan
-	// mitigation.
 	if !instance.Status.AsyncOpInProgress && instance.Status.ReconciledGeneration == 0 {
 		finalizers.Delete(finalizerToken)
 		// Clear the finalizer
 		return c.updateServiceInstanceFinalizers(instance, finalizers.List())
 	}
 
-	// All updates not having a DeletingTimestamp will have been handled above
-	// and returned early. If we reach this point, we're dealing with an update
-	// that's actually a soft delete-- i.e. we have some finalization to do.
-	serviceClass, servicePlan, brokerName, brokerClient, err := c.getServiceClassPlanAndServiceBroker(instance)
-	if err != nil {
+	if err := c.deprovisionServiceInstance(instance); err != nil {
```

Comment:
TODO: Just `err` doesn't give us enough specificity to decide whether we can clear the finalizers or not. An async response will not error, but shouldn't clear the finalizers.